framework/web/Request.php

```diff
@@ -1999,6 +2128,86 @@ private function validateCsrfTokenInternal($clientSuppliedToken, $trueToken)
         return $security->unmaskToken($clientSuppliedToken) === $security->unmaskToken($trueToken);
     }
 
+    /**
+     * {@inheritdoc}
+     * @since 2.1.0
+     */
+    public function getAttributes()
+    {
+        if ($this->_attributes === null) {
+            $this->_attributes = $this->defaultAttributes();
+        }
+        return $this->_attributes;
+    }
+
+    /**
+     * @param array $attributes attributes derived from the request.
+     */
+    public function setAttributes(array $attributes)
```

Comment:
Do we need it public?

Reply:
See https://github.com/yiisoft/yii2/pull/15416#discussion_r158818756